### DISCUSSION TITLE
How to export an app as website

### DISCUSSION THREAD

**Discussion by superlopuh** [marimo-team/marimo#553] [Q&A] [answered]
I have an app that I can run locally, and would like to publish it as a static webpage. I know that there's an open issue to make the Python execute locally in the browser, but for my current use-case that's unnecessary, I'd be happy with just a webpage that displays my app's content statically. Is there any way to do this?

**Comment by akshayka** [comment, marked as answer]
Yes, this can be done. Use the "Export as html" action in the notebook menu:

![image](https://github.com/marimo-team/marimo/assets/1994308/84bd37c5-c462-42c1-8027-bf4524b687f1)

(In case you are wondering: The `Publish as HTML` action lets you publish static HTML to the public web `static.marimo.app` -- published HTML is viewable by anyone, whereas exported HTML is downloaded to your machine.)

I will improve our documentation to explain how to do this. In the future we'll add an option to export from the CLI, but we don't have that yet. Thanks for asking!

**Reply by superlopuh** [reply]
Excellent, thank you!

**Reply by ccxuy** [reply]
maybe conside publish as a dynamic web page, so the data source could update accordingly
